### ISSUE TITLE
ISC001 in case of explicit string concatenation

### ISSUE THREAD

**Issue by spaceby** [astral-sh/ruff#13535]
```python
# ISC001 working
s_1 = 'Hello ' 'world'

# ISC001 not working
s_2 = 'Hello ' + 'world'
```
https://play.ruff.rs/019ff1c5-bc88-4ea4-b664-680efe9dc7bd

I would like ISC001 to work with explicit string concatenation as well.

**Comment by MichaReiser** [comment]
Does https://docs.astral.sh/ruff/rules/explicit-string-concatenation/ work for you?

**Comment by spaceby** [comment]
Do you mean using ISC003 to turn explicit string concatenation into implicit, and then using ISC001 to combine them?
This will not work in the following config:
```
[tool.ruff.lint.flake8-implicit-str-concat]
allow-multiline = false
```

**Comment by MichaReiser** [comment]
Kind of. Taking a step back. What's motivating you for this rule? Do you want to detect any form of string concatenation where both sides are constants? 

My main concern is that wich said rule, the only option to write very long string literals and keep them inside of a reasonable line length is to use a triple-quoted strings.

**Comment by spaceby** [comment]
Ruff has a rule that detects single-line implicit string concatenation [(ISC001)](https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/#single-line-implicit-string-concatenation-isc001)
But ruff does not detect single-line explicit string concatenation
This is strange. This also has a negative impact on the readability of the code.

**Comment by MichaReiser** [comment]
To clarify. Do you only want to ban explicit string concatenation on a single line or in general?

**Comment by spaceby** [comment]
Only in one line. As in ISC001

**Comment by spaceby** [comment]
ISC001 helped me find places where I forgot to add a comma in the list of strings.
This is a useful rule.

**Comment by charliermarsh** [comment]
You want to _ban_ explicit concatenation on a single line only? I don't think we're likely to support that, it's the first time we've heard that request, and implicit concatenations on a single line are a common source of bugs.

**Comment by spaceby** [comment]
It seems like it would require an unreasonable amount of work.
Let's close this issue
